lib/cfg-tree.c

```diff
@@ -212,6 +212,22 @@ log_expr_node_set_children(LogExprNode *self, LogExprNode *children)
   self->children = children;
 }
 
+void
+log_expr_node_subscribe_pipe(LogExprNode *self, LogPipe *pipe)
+{
+  pipe->expr_node = log_expr_node_ref(self);
+}
+
+void
+log_expr_node_unsubscribe_pipe(LogExprNode *self, LogPipe *pipe)
```

Comment:
`subscribe()` and `unsubscribe()` do not just increment/decrement a ref counter, they also set a field in the `LogPipe` instance. Because a `LogPipe` field is modified, shouldn't we move these functions to `logmsg.h`?

`log_pipe_subscribe_expr_node` and `log_pipe_unsubscribe_expr_node` would be their name. What do you think?

Reply:
thanks for the note!
(original implementation was 'closer' to this name, but well... it is better to move to `LogPipe` and call it `attach/detach` instead of `subscribe/unsubscribe`)